### DISCUSSION TITLE
How can I avoid creating many windows with identical names running the same command?

### DISCUSSION THREAD

**Discussion by David-Else** [kovidgoyal/kitty#6748] [Q&A]
I am running a shortcut command in a text editor to spawn a Kitty window to use as a terminal for the document being edited:
```
kitty @ launch --no-response --keep-focus --cwd=current --title=Terminal1
```
If I run that again I get another `Terminal1` window, I want it to do nothing or ideally switch focus to the existing `Terminal1` window. Can I do this without resorting to creating a special kitten script? Thanks!

EDIT I am trying a script but it doesn't work, can anyone help?

```sh
#!/bin/bash
if [ $# -eq 0 ]; then
    echo "No argument provided."
    exit 1
else
    # Check if a window with the same name already exists
    window_id=$(kitty @ ls | jq -r --arg title "$1" '.children[] | .windows[]? | select(.title == $title) | .id')

    if [ -n "$window_id" ]; then
        # If it exists, focus on it
        kitty @ focus-window --match id:$window_id
    else
        # If it doesn't exist, create a new window
        kitty @ launch --no-response --keep-focus --cwd=current --title="$1"
    fi
fi
```

**Comment by kovidgoyal** [comment]
launch always creates a new window. You need to use kitten @ focus-window in a
remote control script. Check the return code and run launch accordingly. 

On Tue, Oct 24, 2023 at 09:37:55AM -0700, David Else wrote:
> I am running a shortcut command in a text editor to spawn a Kitty window to use as a terminal for the document being edited:
> ```
> kitty @ launch --no-response --keep-focus --cwd=current --title=Terminal1
> ```
> If I run that again I get another `Terminal1` window, I want it to do nothing or ideally switch focus to the existing `Terminal1` window. Can I do this without resorting to creating a special kitten script? Thanks!
> 
> -- 
> Reply to this email directly or view it on GitHub:
> https://github.com/kovidgoyal/kitty/discussions/6748
> You are receiving this because you are subscribed to this thread.
> 
> Message ID: ***@***.***>

-- 
_____________________________________

Dr. Kovid Goyal 
https://www.kovidgoyal.net
https://calibre-ebook.com
_____________________________________

**Reply by David-Else** [reply]
Thanks for the feedback! I think I am almost there, but it does not work and gives error when trying to create a window called `example`:

```
...
kitty.search_query_parser.ParseException: No location specified before window_title:example

No location specified before window_title:example
```

```sh
#!/bin/bash
if [ $# -eq 0 ]; then
    echo "No argument provided."
    exit 1
else
    kitty @ focus-window --match window_title:"$1" || kitty @ launch --no-response --keep-focus --cwd=current --title="$1"
fi
```

**Comment by kovidgoyal** [comment]
Read the help of the --match option carefully, there is no such field as
window_title

**Reply by David-Else** [reply]
I got it in the end:
```
    kitty @ focus-window --match title:"^$1" 2>/dev/null || kitty @ launch --no-response --keep-focus --cwd=current --title="$1"
```
Why do I need `^` in the `--match title:`? For a window with the exact title `example`:
```
kitty @ focus-window --match title:^example
```
Works.
```
kitty @ focus-window --match title:example
```
Fails silently with zero feedback to the user.

I can find no mention of this in the help section and seems to make no sense, I am not looking for a window that begins with something, I am looking for a specific window.
